services/examples/request.rs

```diff
@@ -0,0 +1,40 @@
+use geoengine_services::server;
+use tokio::sync::oneshot;
+use tokio::sync::oneshot::Sender;
+
+// Example of a client communicating with the geo engine
+#[tokio::main]
+async fn main() {
+    // TODO: use special config for port etc. for starting the server and connecting to it
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
+
+    let (server, _) = tokio::join!(
+        server::start_server(Some(shutdown_rx)),
+        queries(shutdown_tx),
+    );
```

Comment:
Isn't there a race condition that the server must start before executing `queries`?

Reply:
it worked for me, but your are right. Would it be an acceptable solution to sleep the queries task for some time before issuing the first query or do we need a backchannel from the server that indicates that the server is running? Not sure if warp supports this though...

Reply:
3304dc28317951e2ab2941d70b7815d22fdaca76

Reply:
This works, it's just… couldn't you instead of using `join!`, which does things in parallel, first wait for a response of the server startup and then issue the queries?

Reply:
there is no response from the server. The server is spawned as a tokio task which is then awaited on. This await returns only when the server is shutdown. I haven't found a hook in warp that would allow being notified once the server is ready to serve requests and I think that's because usually a server isn't used from within the server executable it is run in.

Reply:
Hm, but even MAPPING can say: I'm up and running, awaiting requests. Is there an example for that? We could postpone it as a different task though.

Reply:
that's what I'm checking in the wait method. if there is some response from the server.

Reply:
This solution is like pressing F5 in a browser. I'm asking if there is any chance that our server can state internally that it runs.
I guess it can block until it receives incoming requests.
If not, then this is fine.

Reply:
I haven't found anything for querying the server state or even getting notified of successful startup. The documentation of `run` states
> Run this Server forever on the current thread.